src/repository/opamGit.ml

```diff
@@ -205,7 +205,7 @@ module VCS : OpamVCS.VCS = struct
     else if OpamSystem.file_is_empty patch_file then
       (finalise (); Done None)
     else
-      Done (Some (OpamFilename.of_string patch_file))
+      Done (Some (OpamFilename.of_string patch_file, Patch.to_diffs ~p:1 (String.concat "\n" r.r_stdout)))
```

Comment:
the use of `Patch.to_diffs` should be avoided in opam as long as the diff parser isn't complete and well tested.

Instead we can use `OpamRepositoryBackend.get_diff`, ignore all the changes in `.git`/`.hg`/`.darcs` and voilà (hopefully)